mongofiles/options.go

```diff
@@ -22,6 +31,53 @@ Possible commands include:
 
 See http://docs.mongodb.org/manual/reference/program/mongofiles/ for more information.`
 
+// Creates a new ToolOptions and configures it using the command line arguments.
+func ParseOptions(rawArgs []string, storageOpts *StorageOptions, inputOpts *InputOptions) ([]string, *options.ToolOptions, error) {
+	// initialize command-line opts
+	opts := options.New("mongofiles", Usage, options.EnabledOptions{Auth: true, Connection: true, Namespace: false, URI: true})
+
+	opts.AddOptions(storageOpts)
+	opts.AddOptions(inputOpts)
+	opts.URI.AddKnownURIParameters(options.KnownURIOptionsReadPreference)
+	opts.URI.AddKnownURIParameters(options.KnownURIOptionsWriteConcern)
+
+	args, err := opts.ParseArgs(rawArgs)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error parsing command line options: %v\ntry 'mongofiles --help' for more information", err)
+	}
+
+	log.SetVerbosity(opts.Verbosity)
+	signals.Handle()
+
+	// verify uri options and log them
+	opts.URI.LogUnsupportedOptions()
+
+	// add the specified database to the namespace options struct
+	opts.Namespace.DB = storageOpts.DB
+
+	// set WriteConcern
+	if storageOpts.WriteConcern != "" {
+		opts.WriteConcern, err = db.PackageWriteConcernOptsToObject(storageOpts.WriteConcern, nil)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error parsing --writeConcern: %v", err)
+		}
+	}
+
+	// set ReadPreference
+	var readPref *readpref.ReadPref
```

Comment:
nit: is there any reason to use a separate variable rather than just directly setting opts.ReadPreference? (like you did above with WriteConcern.)